esm-loader.mjs

```diff
@@ -92,3 +157,37 @@ export function addESMSupportabilityMetrics(agent) {
     agent.metrics.getOrCreateMetric(NAMES.FEATURES.ESM.UNSUPPORTED_LOADER).incrementCallCount()
   }
 }
+
+/**
+ * Rewrites the source code of a ES module we want to instrument.
+ * This is done by injecting the ESM shim which proxies every property on the exported
+ * module and registers the module with shimmer so instrumentation can be registered properly.
+ *
+ * @param {string} url the URL returned by the resolve chain
+ * @param {string} specifier string identifier in an import statement or import() expression
+ * @returns {string} source code rewritten to wrap with our esm-shim
+ */
+async function wrapEsmSource(url, specifier) {
+  const pkg = await import(url)
+  const props = Object.keys(pkg)
+  const trimmedUrl = url.replace('file://', '')
+
+  return `
+    import wrapModule from '${esmShimPath.href}'
+    import * as _originalModule from '${url}'
+    // lets have as little code in here as possible and push most to
```

Comment:
Are these comments necessary/useful? I feel like since we want "as little code in here as possible", perhaps we should nix them?

Reply:
I was going back and forth on this. i can omit.